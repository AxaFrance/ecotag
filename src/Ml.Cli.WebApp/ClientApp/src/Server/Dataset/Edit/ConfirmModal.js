import React from "react";
import BooleanModal from "@axa-fr/react-toolkit-modal-boolean";

const ConfirmModal = ({isOpen, onCancel, onSubmit}) => {


    return (
        <BooleanModal
            isOpen={isOpen}
            title='Voulez-vous verrouiller le dataset définitivement ?'
            id='modalId'
            onCancel={onCancel}
            onSubmit={onSubmit}
            submitTitle='Confirmer'
            cancelTitle='Annuler'
            className="edit-dataset__modal"
        >
            <p className="edit-dataset__modal-core-text">
                Cette action est définitive. <br />
                Toute modification (ajout, supression de fichier) sera impossible par la suite.
            </p>
        </BooleanModal>
    );
};

export default ConfirmModal;
