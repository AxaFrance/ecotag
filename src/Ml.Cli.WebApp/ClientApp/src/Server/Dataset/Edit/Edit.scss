.edit-dataset {
  &__file-upload-container {
    padding: 0 50px;
    margin-bottom: 50px;
   &--disabled {
     display: none;
   }
 }

  &__file-upload-title {
    text-align: center;
  }

  .af-btn {
    width: 25%
  }

  .af-tabs__control {
    margin-left: 10px;
  }

  &__modal-core-text {
    font-weight: 800;
    font-size: 20px;
    letter-spacing: 1.1px;
  }

 &__file-list-container {
   margin-bottom: 50px;
 }

 &__lock-button-container {
  display: flex;
  justify-content: center;
 }

 &__lock-button-button {
    &--disabled {
      display: none;
    }
 }

  &__lock-button-icon {
    margin-left: 5%;
  }
}
.af-modal__header-title {
  letter-spacing: 1.2px;
  font-size: 31px;
  color: #ef0e0e;
  font-weight: 700;
}
