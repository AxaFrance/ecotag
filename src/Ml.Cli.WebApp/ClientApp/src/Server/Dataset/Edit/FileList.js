import React from 'react';
import Table, {Paging} from '@axa-fr/react-toolkit-table';
import cuid from 'cuid';
import {
    ArticleRestitution,
    SectionRestitution,
    SectionRestitutionColumn,
    SectionRestitutionRow,
    Restitution,
} from '@axa-fr/react-toolkit-restitution';
import Action from "@axa-fr/react-toolkit-action";
import Tabs from "@axa-fr/react-toolkit-tabs/dist";
import '@axa-fr/react-toolkit-tabs/dist/tabs.scss';
import {computeNumberPages, filterPaging} from "../../shared/filtersUtils";
import {formatTimestampToString} from "../../date";
import {resilienceStatus} from "../../shared/Resilience";

const bytesToSize=(bytes) => {
    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
    if (bytes === 0) return '0 Byte';
    const i = parseInt(Math.floor(Math.log(bytes) / Math.log(1024)));
    return Math.round(bytes / Math.pow(1024, i), 2) + ' ' + sizes[i];
}

const downloadAsync = (fetch) => (datasetId, fileId, fileName) => async event => {
    event.preventDefault();
    const response = await fetch(`datasets/${datasetId}/files/${fileId}`, {
        method: 'GET'
    });
    const blob = await response.blob();
    const url = window.URL.createObjectURL(blob);
    const a = document.createElement('a');
    a.href = url;
    a.download = fileName;
    document.body.appendChild(a);
    a.click();
    a.remove();         
};


const FileList = ({state, setState, fetch}) => {

    const deleteFile = async file => {
        setState({...state, status: resilienceStatus.POST});

        const response = await fetch(`datasets/${state.dataset.id}/files/${file.file.id}`, {
            method: 'DELETE'
        });
        
        if(response.status >= 500){
            setState({...state, status : resilienceStatus.ERROR });
            return;
        }

        const filesSend = [...state.files.filesSend];
        const index = filesSend.indexOf(file);
        if (index > -1) {
            filesSend.splice(index, 1);
        }
        setState({...state, files: {...state.files, filesSend}, status: resilienceStatus.SUCCESS});
    };

    const onChangePaging = ({numberItems, page}) => {
        setState({...state,
            files : {...state.files, paging: {
                itemByPages: numberItems,
                currentPages: page,
        }}});
    };
    const files = state.files;
    const paging = files.paging;
    const itemByPages = paging.itemByPages
    const numberPages = computeNumberPages(files.filesSend, itemByPages);
    const currentPages = paging.currentPages > numberPages ? numberPages :  paging.currentPages
    const itemFiltered= filterPaging(files.filesSend, itemByPages, currentPages);

    const reducer = (previousValue, currentValue) => previousValue + currentValue.file.size;
    const hasFiles = state.files.filesSend.length === 0;
    const fileSizeTotal =  hasFiles? 0 : state.files.filesSend.reduce(reducer, 0);
    const fileSizeAverage = hasFiles? 0 : fileSizeTotal / state.files.filesSend.length;

    return (
        <div className="edit-dataset__file-list-container">
            <Tabs classModifier="container" onChange={() => {}}>
                <Tabs.Tab title="Information générales">
                    <ArticleRestitution>
                        <SectionRestitution>
                            <SectionRestitutionRow title="">
                                <SectionRestitutionColumn>
                                    <Restitution label="Nom" value={state.dataset.name} />
                                    <Restitution label="Date de création" value={formatTimestampToString(state.dataset.createDate)} />
                                    <Restitution label="Type" value={state.dataset.type} />
                                </SectionRestitutionColumn>
                                <SectionRestitutionColumn>
                                    <Restitution label="Nombre de fichier" value={state.files.filesSend.length} />
                                    <Restitution label="Poids total des fichiers" value={bytesToSize(fileSizeTotal)} />
                                    <Restitution label="Poids moyen des fichiers" value={bytesToSize(fileSizeAverage)} />
                                </SectionRestitutionColumn>
                            </SectionRestitutionRow>
                        </SectionRestitution>
                    </ArticleRestitution>
                </Tabs.Tab>
                <Tabs.Tab title="Liste des fichiers">
                    <Table>
                        <Table.Header>
                            <Table.Tr>
                                <Table.Th>
                                    <span className="af-table__th-content">Nom</span>
                                </Table.Th>
                                <Table.Th>
                                    <span className="af-table__th-content ">Type</span>
                                </Table.Th>
                                <Table.Th>Size</Table.Th>
                                {state.dataset.isLock ? null :<Table.Th>Action</Table.Th>}
                            </Table.Tr>
                        </Table.Header>
                        <Table.Body>
                            {itemFiltered.map(file => (
                                    <Table.Tr key={cuid()}>
                                        <Table.Td>
                                         <a href={"#"+file.file.name} alt={file.file.name} onClick={downloadAsync(fetch)(state.dataset.id, file.file.id, file.file.name)}>{file.file.name}</a> 
                                        </Table.Td>
                                        <Table.Td>{file.file.type}</Table.Td>
                                        <Table.Td>{bytesToSize(file.file.size)}</Table.Td>
                                        {state.dataset.isLock ? null :<Table.Td> <Action
                                            id="deleteButton"
                                            icon="trash"
                                            title="Supprimer"
                                            onClick={() => deleteFile(file)}
                                        /></Table.Td>}
                                    </Table.Tr>
                                ))}
                        </Table.Body>
                    </Table>
                    <Paging
                        onChange={onChangePaging}
                        numberItems={itemByPages}
                        numberPages={numberPages}
                        currentPage={currentPages}
                        id="home_paging"
                    />
                </Tabs.Tab>
            </Tabs>
        </div>
    )
}

export default FileList;
