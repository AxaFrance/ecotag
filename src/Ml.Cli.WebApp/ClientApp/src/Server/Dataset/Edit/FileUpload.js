import React from 'react';
import { File, FileTable } from '@axa-fr/react-toolkit-form-input-file';
import Button from '@axa-fr/react-toolkit-button';
import cuid from 'cuid';
import {resilienceStatus} from "../../shared/Resilience";

export const FileUpload = ({fetch, setState, state}) => {

    const onChange = value => {
        const reader = new FileReader();
        reader.onloadend = () => {
            setState({ ...state, files : { ...state.files, filesLoad: value }});
        };
        reader.readAsDataURL(value.values[0].file);
    };

    const deleteFile = (idFile) => {
        const files = state.files;
        const filesLoad = files.filesLoad;
        const file = filesLoad.values.find(v => v.id === idFile);
        const index = filesLoad.values.indexOf(file);
        const newValues = [...filesLoad.values];
        if (index > -1) {
            newValues.splice(index, 1);
        }
        setState({ ...state, files : { ...state.files, filesLoad: { values : newValues }  }});
    };

    const sendFile = async() => {

        setState({...state, status : resilienceStatus.POST });
     
            let i, j, temporary, chunk = 10;
            const array = state.files.filesLoad.values;
            const promises = [];
            for (i = 0,j = array.length; i < j; i += chunk) {
                temporary = array.slice(i, i + chunk);
                const formData = new FormData();
                for (const value of temporary) {
                    formData.append('files', value.file);
                }
                const responsePromise = fetch(`datasets/${state.dataset.id}/files`, {
                    method: 'POST',
                    header: {
                        'content-type': 'multipart/form-data',
                    },
                    body: formData,
                });
                promises.push(responsePromise)
            }

        const responses = await Promise.all(promises);
            if(responses.find(response => response.status >= 500)){
                setState({...state, status : resilienceStatus.ERROR });
            } else {
                const newFilesSend = state.files.filesSend;
                state.files.filesLoad.values.map(file => {
                    const newFile = {...file, id: cuid()};
                    newFilesSend.push(newFile);
                });
                setState({...state, files: {...state.files, filesLoad: [], filesSend: newFilesSend}});
            }
    };

    return (
        <>
            <div className={`edit-dataset__file-upload-container edit-dataset__file-upload-container--${state.dataset.isLock ? 'disabled' : ''}`}>
                <h2 className="edit-dataset__file-upload-title">Upload des fichiers {state.dataset.type === "Image" ? '.jpg, .jpeg, .png, .tiff': '.txt'}</h2>
                <File
                    id='file'
                    name='datasetUploadFiles'
                    onChange={onChange}
                    multiple={true}
                    isVisible={true}
                    readOnly={false}
                    disabled={false}
                    accept={state.dataset.type === "Image" ? 'image/jpeg, image/png, image/tiff': 'text/plain'}
                    label="Parcourir"
                    icon='open'
                />
                {state.files.filesLoad.length === 0 ? '' : <FileTable errors={[]} values={state.files.filesLoad.values} onClick={deleteFile} />}
                <div>
                    <Button
                        name="sendFiles"
                        id="sendFiles"
                        onClick={sendFile}
                        disabled={state.files.filesLoad.length === 0}
                        classModifier={state.files.filesLoad.length === 0 ? 'disabled' : ''}>
                        <span className="af-btn__text">Envoyer</span>
                    </Button>
                </div>
            </div>
        </>
    );
};

export default FileUpload;
