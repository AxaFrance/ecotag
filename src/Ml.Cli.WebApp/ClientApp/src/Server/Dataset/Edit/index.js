export { default } from './EditContainer.js';
