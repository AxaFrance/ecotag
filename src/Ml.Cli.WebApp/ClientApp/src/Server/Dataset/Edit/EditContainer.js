import React, {useState} from 'react';
import FileUpload from "./FileUpload";
import FileList from "./FileList";
import Lock from "./Lock";
import ConfirmModal from "./ConfirmModal";
import './Edit.scss';
import Title from "../../shared/Title/Title.container";
import {computeNumberPages, filterPaging} from "../../shared/filtersUtils";


export const EditContainer = () => {
    const [state, setState] = useState({
        filesLoad: [],
        filesSend: [],
        fileData: null,
        openLockModal: false,
        isLock: false,
        paging: {
            numberPages: computeNumberPages([], 10),
            currentPages: 1,
            itemByPages: 5,
            itemFiltered: filterPaging([], 10, 1)
        }
    });

    return (
        <div className="edit-dataset">
            <ConfirmModal state={state} setState={setState}/>
            <Title back>Edition du dataset</Title>
            <FileUpload state={state} setState={setState}/>
            {state.filesSend.length === 0 ? null : <FileList state={state} setState={setState} />}
            <Lock state={state} setState={setState} />
        </div>
    );
};

export default EditContainer;
