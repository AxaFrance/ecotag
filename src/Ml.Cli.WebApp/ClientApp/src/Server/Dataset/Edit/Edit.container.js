import React, {useState} from 'react';
import FileUpload from "./FileUpload";
import FileList from "./FileList";
import './Edit.scss';
import Title from "../../../TitleBar";
import {computeNumberPages, filterPaging} from "../../shared/filtersUtils";
import {fetchDataset, fetchLockDataset} from "../Dataset.service";
import {resilienceStatus, withResilience} from "../../shared/Resilience";
import {useParams} from "react-router-dom";
import withCustomFetch from "../../withCustomFetch";
import Lock from "../../shared/Lock/Lock";
import ConfirmModal from "../../shared/ConfirmModal/ConfirmModal";

export const init = (fetch, setState) => async (id, state) => {
    const response = await fetchDataset(fetch)(id);
    let data;
    if(response.status >= 500) {
        data = { status: resilienceStatus.ERROR };
    } else {
        const dataset = await response.json()
        const datasetData = { name: dataset.name, 
            id: dataset.id, 
            createDate: dataset.createDate,
            isLock: dataset.isLocked,
            type: dataset.type
        };
        const filesSend= dataset.files.map( f => { return  { file: { id: f.id, type: f.contentType, size: f.size, name: f.fileName } }})
        data = { status: resilienceStatus.SUCCESS, dataset:datasetData, files : {...state.files, filesSend} };
    }
    setState({ ...state, ...data});
};

const lockDataset = (fetch, setState) => async (state, idDataset) => {
    const response = await fetchLockDataset(fetch)(idDataset);
    let data;
    if (response.status >= 500) {
        data = {status: resilienceStatus.ERROR};
    } else {
        data = {status: resilienceStatus.SUCCESS};
    }
    setState({...state, ...data, openLockModal: false, dataset: {...state.dataset, isLock:true }});
};

export const Edit = ({fetch, state, setState, lock}) => {
    const isDisabled = state.files.filesSend.length === 0;
    return(
        <div className="edit-dataset">
            <ConfirmModal title='Voulez-vous verrouiller le dataset définitivement ?' isOpen={state.openLockModal} onCancel={lock.onCancel} onSubmit={lock.onSubmit}>
                <p className="edit-dataset__modal-core-text">
                    Cette action est définitive. <br />
                    Toute modification (ajout, supression de fichier) sera impossible par la suite.
                </p>
            </ConfirmModal>
            <Title title={state.dataset.name} subtitle="Edition du dataset" goTo="/datasets" goTitle="Datasets"/>
            <FileUpload fetch={fetch} state={state} setState={setState}/>
            {state.files.filesSend.length === 0 ? null : <FileList fetch={fetch} state={state} setState={setState}/>}
            <Lock text="Verrouiller" lockedText="Dataset verrouillé" isDisabled={isDisabled} isLocked={state.dataset.isLock} onLockAction={lock.onLockDataset}/>
        </div>
    );
}

const PageWithResilience = withResilience(Edit);

export const EditContainer = ({fetch}) => {
    const { id } = useParams();
    const [state, setState] = useState({
        dataset: { name: "", id: "", createDate:null, isLock:false },
        status: resilienceStatus.LOADING,
        files:{
            filesLoad: { values : [] },
            filesLoadError: [],
            filesSend: [],
            filesSendError: [],
            paging: {
                numberPages: computeNumberPages([], 10),
                currentPages: 1,
                itemByPages: 5,
                itemFiltered: filterPaging([], 10, 1)
            }
        },
        openLockModal: false,
    });

    
    const lock = {
        onLockDataset: () =>{
            if(state.dataset.isLock){
                return;
            }
            setState({...state, openLockModal: true});
        },
        onCancel : () => {
            setState({...state, openLockModal: false});
        },

        onSubmit : () => {
            lockDataset(fetch, setState)(state, id)
        }
    }

    React.useEffect(() => {
        init(fetch, setState)(id, state);
    }, []);

    return <PageWithResilience fetch={fetch} status={state.status} state={state} setState={setState} lock={lock} />;
};

export default withCustomFetch(fetch)(EditContainer);
