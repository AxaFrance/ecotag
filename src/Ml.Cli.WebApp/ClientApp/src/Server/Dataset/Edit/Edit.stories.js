import React from 'react';
import EditContainer from "./EditContainer";
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

const history = createMemoryHistory();

export default {
    title: 'Edition dataset',
    component: EditContainer
};

const Template = () => <Router history={history}><EditContainer /></Router>;
export const EditDataset = Template.bind({});
