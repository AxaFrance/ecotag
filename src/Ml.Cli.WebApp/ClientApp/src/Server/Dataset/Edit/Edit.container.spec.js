import {render, waitFor} from '@testing-library/react';
import {Edit, EditContainer} from "./Edit.container";
import {BrowserRouter as Router} from "react-router-dom";

describe('Edit.container', () => {
    describe('init', () => {
        it('should init correctly', async () => {
            const fetch = () => {
                return {
                    status: 200,
                    json: () => Promise.resolve({id: "0001", name: "testName", createDate: Date.now(), isLocked: true, type: "image", files: []})
                };
            };
            const {asFragment, getByText} = render(<Router><EditContainer fetch={fetch}/></Router>);
            await waitFor(() => expect(getByText(/Verrouiller/i)));
            expect(asFragment()).toMatchSnapshot();
        });
        it('should init with forbidden message', async () => {
            const fetch = () => {
                return {
                    status: 403,
                };
            };
            const {asFragment, getByText} = render(<Router><EditContainer fetch={fetch}/></Router>);
            await waitFor(() => expect(getByText(/Vous n'avez pas le droit d'accéder à cette ressource./i)));
            expect(asFragment()).toMatchSnapshot();
        });
        it('should init with error message', async () => {
            const fetch = () => {
                return {
                    status: 500,
                };
            };
            const {asFragment, getByText} = render(<Router><EditContainer fetch={fetch}/></Router>);
            await waitFor(() => expect(getByText(/problème/i)));
            expect(asFragment()).toMatchSnapshot();
        });
    });
});

describe('Edit', () => {
    const state = {
        files: {
            filesSend: [],
            filesSendError: [],
            filesLoad: [],
            filesLoadError: [],
            paging: {itemByPages: 50, currentPages: 1}
        },
        openLockModal: false,
        dataset: {
            id: '0001',
            name: 'datasetName',
            createDate: 0,
            type: 'Classification',
            isLock: false
        }
    };

    const lock = {
        onSubmit: () => {},
        onCancel: () => {},
        onLockDataset: () => {}
    };

    it('should render correctly', () => {
        const {asFragment} = render(<Router><Edit state={state} setState={() => {}} fetch={() => {}} lock={lock}/></Router>);
        expect(asFragment()).toMatchSnapshot();
    });
});
