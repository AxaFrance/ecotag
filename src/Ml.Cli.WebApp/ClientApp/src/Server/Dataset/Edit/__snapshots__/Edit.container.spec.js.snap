// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit should render correctly 1`] = `
<DocumentFragment>
  <div
    class="edit-dataset"
  >
    <div
      class="af-title-bar af-title-bar--backhome"
    >
      <div
        class="container-fluid container af-title-bar__wrapper"
      >
        <a
          class="btn af-btn--circle"
          href="/datasets"
          title="Datasets"
        >
          <i
            class="glyphicon glyphicon-arrowthin-left"
          />
        </a>
        <h1
          class="af-title-bar__title"
        >
          datasetName
        </h1>
        <small
          class="af-title-bar__subtitle"
        >
          Edition du dataset
        </small>
      </div>
    </div>
    <div
      class="edit-dataset__file-upload-container edit-dataset__file-upload-container--disabled"
    >
      <h2
        class="edit-dataset__file-upload-title"
      >
        Upload des fichiers .eml
      </h2>
      <div
        class="af-form__file-input"
      >
        <div
          class="drop-box hidden-mobile"
          id="file"
          tabindex="0"
        >
          <input
            accept="message/rfc822,application/octet-stream"
            autocomplete="off"
            multiple=""
            name="datasetUploadFiles"
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <div>
            Glissez/déposez vos fichiers
          </div>
        </div>
        <button
          class="af-btn af-btn--file af-btn--hasIconLeft"
          type="button"
        >
          <i
            class="glyphicon glyphicon-open"
          />
           Parcourir
        </button>
      </div>
      <div>
        <button
          class="btn af-btn af-btn--disabled"
          disabled=""
          id="sendFiles"
          name="sendFiles"
        >
          <span
            class="af-btn__text"
          >
            Envoyer
          </span>
        </button>
      </div>
    </div>
    <div
      class="lock__button-container"
    >
      <button
        class="btn af-btn af-btn--disabled"
        disabled=""
        icon="lock"
        id="lock-button"
      >
        <span
          class="af-btn__text"
        >
          Dataset en cours d'import
        </span>
        <i
          class="lock__button-icon glyphicon glyphicon-lock"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Edit.container init should init correctly 1`] = `
<DocumentFragment>
  <div
    class="edit-dataset"
  >
    <div
      class="af-title-bar af-title-bar--backhome"
    >
      <div
        class="container-fluid container af-title-bar__wrapper"
      >
        <a
          class="btn af-btn--circle"
          href="/datasets"
          title="Datasets"
        >
          <i
            class="glyphicon glyphicon-arrowthin-left"
          />
        </a>
        <h1
          class="af-title-bar__title"
        >
          testName
        </h1>
        <small
          class="af-title-bar__subtitle"
        >
          Edition du dataset
        </small>
      </div>
    </div>
    <div
      class="edit-dataset__file-upload-container edit-dataset__file-upload-container--disabled"
    >
      <h2
        class="edit-dataset__file-upload-title"
      >
        Upload des fichiers .eml
      </h2>
      <div
        class="af-form__file-input"
      >
        <div
          class="drop-box hidden-mobile"
          id="file"
          tabindex="0"
        >
          <input
            accept="message/rfc822,application/octet-stream"
            autocomplete="off"
            multiple=""
            name="datasetUploadFiles"
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <div>
            Glissez/déposez vos fichiers
          </div>
        </div>
        <button
          class="af-btn af-btn--file af-btn--hasIconLeft"
          type="button"
        >
          <i
            class="glyphicon glyphicon-open"
          />
           Parcourir
        </button>
      </div>
      <div
        class="custom-table-file af-file-table"
      />
      <div>
        <button
          class="btn af-btn"
          id="sendFiles"
          name="sendFiles"
        >
          <span
            class="af-btn__text"
          >
            Envoyer
          </span>
        </button>
      </div>
    </div>
    <div
      class="lock__button-container"
    >
      <button
        class="btn af-btn af-btn--disabled"
        disabled=""
        icon="lock"
        id="lock-button"
      >
        <span
          class="af-btn__text"
        >
          Dataset en cours d'import
        </span>
        <i
          class="lock__button-icon glyphicon glyphicon-lock"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Edit.container init should init with error message 1`] = `
<DocumentFragment>
  <div
    class="resilience"
  >
    <div
      class="resilicience__error-message"
    >
      <p>
        Une erreur serveur est survenue.
      </p>
      <br />
      <p>
        Nous sommes au courant du problème. Une investigation est en cours...
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Edit.container init should init with forbidden message 1`] = `
<DocumentFragment>
  <div
    class="resilience"
  >
    <div
      class="resilicience__error-message"
    >
      <p>
        Vous n'avez pas le droit d'accéder à cette ressource.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
